### PR TITLE
[charts] Prepare the charts release

### PR DESCRIPTION
**Pull request by alexfauquette**
Make scripts ready for the release of alpha 0

- Add a changelog text (I was not super inspired might redo it)
- make package public
- update changelog generator script
- make the docs visible

### Changelog

We release a new package `@mui/x-charts`.
This package aims at simplifying the integration of charts into your dashboards. 📊

Screenshot from 2023-06-22 10-29-02

It already contains [lines](https://mui.com/x/react-charts/lines/), [bars](https://mui.com/x/react-charts/bars/), and [scatter](https://mui.com/x/react-charts/scatter/) charts with basic customization features.
Please, visit the new [documentation](https://mui.com/x/react-charts/) to know what it does, and open issues to get the feature you need to be implemented 🚀

### CODE REVIEW

**Comment by joserodolfofreitas** [comment]
Are we not shipping the cool palletes?
We could highlight them on the styling section.

**Comment by flaviendelangle** [comment]
> It already contains lines, bars, and scatter charts with basic customization features.

How about using links for each of those charts (`[lines](https://mui.com/x/react-charts/lines/)`)

**Review by flaviendelangle — APPROVED** [review]
Looks great :smiley: !

File: docs/data/pages.ts [hunk 1]
```diff
@@ -7,10 +7,10 @@ import dataGridComponentApi from './data-grid-component-api-pages';
...
 import chartsComponentApi from './charts-component-api-pages';
 
-const isPreview =
-  process.env.NODE_ENV === 'development' ||
-  process.env.CONTEXT === 'deploy-preview' ||
-  process.env.CONTEXT === 'branch-deploy';
+// const isPreview =
+//   process.env.NODE_ENV === 'development' ||
+//   process.env.CONTEXT === 'deploy-preview' ||
+//   process.env.CONTEXT === 'branch-deploy';
```
Comment (oliviertassinari) [inline docs/data/pages.ts:13]: `process.env.DEPLOY_ENV` already exists for this use case. Removed in 8757e19e01795eacd190a6027ef7bb2caccfa9bc.

**Pull request merged**